token/program-2022/src/pod.rs

```diff
@@ -95,16 +110,69 @@ impl From<OptionalNonZeroElGamalPubkey> for Option<ElGamalPubkey> {
         }
     }
 }
+#[cfg(feature = "serde-traits")]
+impl Serialize for OptionalNonZeroElGamalPubkey {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        if self.0 == ElGamalPubkey::default() {
+            s.serialize_none()
+        } else {
+            s.serialize_some(&self.0.to_string())
+        }
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+struct OptionalNonZeroElGamalPubkeyVisitor;
+#[cfg(feature = "serde-traits")]
+impl<'de> Visitor<'de> for OptionalNonZeroElGamalPubkeyVisitor {
+    type Value = OptionalNonZeroElGamalPubkey;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an ElGamal public key as base64")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+    {
+        let bytes = BASE64_STANDARD.decode(v).map_err(de::Error::custom)?;
+
+        if bytes.len() != 32 {
+            return Err(de::Error::custom("Length of base64 decoded bytes is not 32"));
+        }
+
+        let mut array = [0; 32];
+        array.copy_from_slice(&bytes[0..32]);
+
+        let elgamal_pubkey = ElGamalPubkey(array);
```

Comment:
Do you need this copy here? Or can we just do `ElGamalPubkey::from_bytes(&bytes[..32])`?

Reply:
`pod::ElGamalPubkey` does not implement `from_bytes` but `decoded::ElGamalPubkey` (the non-POD ElGamalPubkey type) does. So I could use `decoded::ElGamalPubkey::from_bytes()` to create a new `decoded::ElGamalPubkey` instance, and then finally convert that to `pod::ElGamalPubkey`. I could do that, but `decoded::ElGamalPubkey::from_bytes()` also uses a copy, so the benefit of doing it might not be there.

Note that I move the code your comment was made under here: token/program-2022/src/serialization.rs in `fn base64_to_bytes`